main.py

```diff
@@ -1,22 +1,37 @@
-from flask import Flask, request
-from flask_json import FlaskJSON
+import json
+from flask import Flask, request, make_response
 
 app = Flask(__name__)
-FlaskJSON(app)
 
-@app.route("/")
-def hello():
-    return "Hello World!"
+class ContentError(Exception):
+    pass
+class LogError(Exception):
+    pass
 
 @app.route('/login', methods=['POST'])
 def login():
+    try:
+        if request.headers['Content-Type'] == 'application/json':
+            data = json.loads(request.get_data())
+            if 'name' and 'password' in data.keys():
+                if data['name'] == 'denchik' and data['password'] == 'foobar':
+                    message = 'OK'
+                    http_code = 200
+                else:
+                    raise LogError
+            else: 
+                raise ContentError
+        else:
+            raise ContentError
+    except ContentError:
+        message = 'BAD REQUEST'
```

Comment:
The text returned in this and other cases duplicates the meaning if the HTTP code, it's ok to not send anything in the response body if the text is unambiguous. It's also good to explain *why* the request failed ("the request body is not JSON", the request is missing a key etc.)